arch/arm/boot/dts/qcom/qcom-msm8974-htc-m8.dts

```diff
@@ -0,0 +1,332 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include "qcom-msm8974.dtsi"
+#include "qcom-pm8841.dtsi"
+#include "qcom-pm8941.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "HTC One (M8)";
+	compatible = "htc,m8", "qcom,msm8974";
+	chassis-type = "handset";
+
+	aliases {
+		mmc0 = &sdhc_1;
+	};
+
+	chosen {
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_keys_default>;
+
+		key-volume-down {
+			label = "volume_down";
+			gpios = <&tlmm 27 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEDOWN>;
+			wakeup-source;
+			debounce-interval = <15>;
+		};
+
+		key-volume-up {
+			label = "volume_up";
+			gpios = <&tlmm 28 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEUP>;
+			wakeup-source;
+			debounce-interval = <15>;
+		};
+	};
+};
+
+&pm8941_vib {
+	status = "okay";
+};
+
+&pronto {
+	/* Use v1 to get correct vddmx-supply range. */
+	compatible = "qcom,pronto-v1-pil", "qcom,pronto";
```

Comment:
Can you link to downstream devicetree why this is necessary? Does your device maybe come with a different WiFi/Bluetooth chip than WCN3680?

Reply:
Sure, please have a look [here](https://github.com/LineageOS/android_kernel_htc_msm8974/blob/lineage-18.1/drivers/net/wireless/wcnss/wcnss_vreg.c#L115).

I did not really investigate why the mismatch. When I saw that `pronto-v1-pil` matched the vendor kernel, I went with it.

According to the iFixit teardown, it is indeed a WCN3680, maybe an older/different revision?

Reply:
Does not overriding the compatible also work? I checked a bit and other 8974 devices seem to look the same as your M8

Reply:
I tried to keep the default compatible, and wifi does work. But I am seeing an error message in the logs: `s1: unsupportable voltage range: 1287500-1050000uV` which is probably why I investigated this and found that the `pronto-v1-pil` had a correct voltage range, same range as the vendor kernel.

Since it works with the compatible from the dtsi, I removed the override here.

Reply:
> s1: unsupportable voltage range: 1287500-1050000uV

Oh that's where this error is from :sweat_smile: I've seen this since forever on FP2 also but it's not a fatal error :) 

I'll make an issue for fixing this for all, since it should be applicable to all msm8974 devices with pronto. https://github.com/msm8974-mainline/linux/issues/15